codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -399,7 +365,20 @@ public Http2Stream close() {
             }
 
             state = CLOSED;
-            deactivateInternal(this);
+            if (activeStreams.remove(this)) {
+                try {
+                    // Update the number of active streams initiated by the endpoint.
+                    createdBy().numActiveStreams--;
+
+                    // Notify the listeners.
+                    for (Listener listener : listeners) {
+                        listener.streamClosed(this);
```

Comment:
Agreed, I think we should address this as a separate issue ... created #3501. 

WRT the finally block, I think I'd like to keep it in this PR since not doing this could lead to a memory leak if a listener throws.